src/core_modules/capture-core/components/Pages/EnrollmentEditEvent/EnrollmentEditEventPage.container.js

```diff
@@ -44,24 +46,38 @@ const getPageStatus = ({ orgUnitId, enrollmentSite, teiDisplayName, trackedEntit
 };
 
 export const EnrollmentEditEventPage = () => {
-    const { orgUnitId, eventId } = useLocationQuery();
+    const history = useHistory();
+    const dispatch = useDispatch();
+
+    const eventId = useSelector(({ viewEventPage }) => viewEventPage.eventId);
+    const error = useSelector(({ activePage }) => activePage.viewEventLoadError?.error);
     const { loading, event } = useEvent(eventId);
     const { program: programId, programStage: stageId, trackedEntity: teiId, enrollment: enrollmentId } = event;
+    const { orgUnitId, eventId: urlEventId } = useLocationQuery();
 
-    return !loading && eventId ? (
+    useEffect(() => {
+        if (!urlEventId) {
+            // return to main page
+            history.push(`/?${buildUrlQueryString({ orgUnitId })}`);
+        } else if (eventId !== urlEventId) {
+            dispatch(changeEventFromUrl(urlEventId, pageKeys.ENROLLMENT_EVENT));
+        }
+    }, [dispatch, history, eventId, urlEventId, orgUnitId]);
+
+    return !loading && eventId === urlEventId || error ? (
```

Comment:
IMHO the mix of `&&` and `||` is a bit confusing. Can you add some parentheses to clarify the intended order of operations? 
Thank you!